### DISCUSSION TITLE
How to create this Quasar/QDate effect using pure nicegui?

### DISCUSSION THREAD

**Discussion by Nemecsek** [zauberzeug/nicegui#2880] [Q&A]
### Question

I already posted on [Reddit](https://www.reddit.com/r/nicegui/comments/1c25jdb/how_to_create_this_quasarqdate_effect_using_pure/) this request 

I would like to create the nice date input as shown in [Quasar/QDate](https://quasar.dev/vue-components/date/#with-additional-buttons). If it doesn't jump to the proper anchor in the page, please click on the title bar and confirm it with enter. 

The zooming popup effect is not important, even if nice to see: instantaneous show/hide are ok, 
![Peek 2024-04-12 12-05](https://github.com/zauberzeug/nicegui/assets/25978536/68d42e6b-c10d-480d-a1cb-cd18b8255161)

I don't know how to code in [ui.date](https://ui.date/)'s props Quasar's options to obtain it.
Is it possible?

SIDE QUESTION: Where is the best place to ask for help about NiceGUI? Posting multiple times on Reddit, Discord, Github Issues, Github discussions, is far from being efficient. Why not to move the community to a single place chosen by the developers?

**Comment by falkoschindler** [comment]
Hi @Nemecsek,

Have you seen https://nicegui.io/documentation/date#input_element_with_date_picker? If you don't need the input element, you can just use the icon and the menu.

---

Yes, there are quite a lot of places to ask questions about NiceGUI.
- We definitely need GitHub for development and issue tracking. That's the place we spend most of our time and where the community can reach us maintainers.
- Discord has been requested by the community (https://github.com/zauberzeug/nicegui/discussions/550) as a place to discuss with the community, without generating too much noise on GitHub.
- Of course there's also StackOverflow, even though NiceGUI users don't seem to be very active there.
- Last but not least, Reddit is a platform where we mainly announce new features and answer questions occasionally.

I'm not sure if we could discard any of them. We definitely need GitHub, the community wants Discord, StackOverflow is there anyway, and for some users Reddit might be the platform of choice to get news about new releases.

As a rule of thumb: To reach the maintainers and developers, GitHub is the right place. If you primarily want to address other users, you can choose Discord, StackOverflow or Reddit - or ChatGPT!

**Reply by Nemecsek** [reply]
Hi @falkoschindler. Thank you for your answer. 
I already gave a look to the link you provide but the behavior of the demo in the page is different from what I get if I copy the example code of **Input element with date picker**.

1) In the example after the click it closes the menu automatically. In my case after choosing the day **I need to click outside the menu in order to close it**. (I opened [Issue #2896](https://github.com/zauberzeug/nicegui/issues/2896) to keep track of this.

2) I need to use a **range** of dates. 
When you return a single day, it is a string that can be **bind_value**d.
If you return **a range it is a dictionary "from", "to"**, binding the value will read **[Object, Object]** and not the date string in the format "DD/MM/YYYY - DD/MM/YYYY" or any other standard format.

**Reply by falkoschindler** [reply]
Hi @Nemecsek, regarding a range of dates: You can use a conversion function to define how to translate the dictionary into a string.
```py
data = ui.date().props('range')
ui.label().bind_text_from(data, 'value', lambda d: f'from {d["from"]} to {d["to"]}' if d else 'no range selected')
```

**Reply by Nemecsek** [reply]
@falkoschindler , I would temporarily suspend this Discussion and continue on [issue 2896](https://github.com/zauberzeug/nicegui/issues/2896) otherwise it is difficult to follow both at the same time. At the end I will link them both reciprocally to avoid losing them for posterity.
